src/components/PostDetails.tsx

```diff
@@ -5,31 +5,39 @@
 import * as commentsApi from '../api/comments';
 
 import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
+import { CommentData } from '../types/Comment';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { actions as commentsActions } from '../features/comments/comments';
 
 type Props = {
   post: Post;
 };
 
 export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
   const [visible, setVisible] = useState(false);
 
+  const dispatch = useAppDispatch();
+  const { items, loaded, hasError } = useAppSelector(state => state.comments);
+
   function loadComments() {
-    setLoaded(false);
-    setError(false);
+    dispatch(commentsActions.setCommentLoader(false));
```

Comment:
The loader is set to `false` before fetching comments, which might be incorrect. Typically, you want to set the loader to `true` to indicate loading has started.